docs/source/architecture/index.rst

```diff
@@ -0,0 +1,122 @@
+On the basis of transactions that have passed stateful validation **verified proposal** will be created and passed to `Block Creator <#block-creator>`__.
+All the transactions that have not passed stateful validation will be dropped and not included in a verified proposal.
+
+Block Creator
+-------------
+
+System component that forms a block from a set of transactions that have passed `stateless <../core_concepts/glossary.html#stateless-validation>`__ and `stateful <../core_concepts/glossary.html#stateful-validation>`__ validation for further propagation to `consensus <#consensus>`__.
+
+Block creator, together with the `Verified Proposal Creator <#verified-proposal-creator>`_ form a component called `Simulator <https://github.com/hyperledger/iroha/tree/master/irohad/simulator>`_.
+
+Consensus
+---------
+
+A consensus algorithm is a process in computer science used to achieve agreement on a single data value among distributed processes or systems.
+Consensus algorithms are designed to achieve reliability in a network involving multiple unreliable nodes.
+Solving that issue -- known as the consensus problem -- is important in distributed computing and multi-agent systems.
+
+*Consensus, as an algorithm*
+
+    An algorithm to achieve agreement on a block among peers in the network. By having it in the system, reliability is increased.
+
+*Consensus, as a component*
+
+    Consensus is the heart of the blockchain - it preserves a consistent state among the `peers <#peer>`__ within a peer network.
+    Iroha uses own consensus algorithm called Yet Another Consensus (aka YAC). 
+
+    You can check out a video where HL Iroha maintainer thoroughly explains the principles of consensus and YAC in particular `here <https://youtu.be/mzuAbalxOKo>`__.
+
+    Distinctive features of YAC algorithm are its scalability, performance and `Byzantine fault tolerance <https://en.wikipedia.org/wiki/Byzantine_fault_tolerance>`_.
+
+    To ensure consistency in the network, if there are missing blocks, they will be downloaded from another peer via `Synchronizer <#synchronizer>`__.
+    Committed blocks are stored in `Ametsuchi <#ametsuchi>`__ block storage.
+
+
+Synchronizer
+------------
+
+Is a part of `consensus <#consensus>`__.
+Adds missing blocks to `peers' <../core_concepts/glossary.html#peer>`__ chains (downloads them from other peers to preserve consistency).
+
+Ametsuchi Blockstore
+--------------------
+
+Iroha storage component, which stores blocks and a state generated from blocks, called `World State View <../core_concepts/glossary.html#world-state-view>`__.
+There is no way for the `client <../core_concepts/glossary.html#client>`__ to directly interact with Ametsuchi.
+
+
```

Comment:
Not sure if so many empty lines are required in the end of the file. 👀

Reply:
🤦